src/Services/CourseService.php

```diff
@@ -62,4 +65,40 @@ public function getCoursesListByCriteria(array $criteria, ?PaginationDto $pagina
         return $courses;
     }
 
+    /**
+     * @param Course $course
+     * @param array $categories
+     */
+    public function attachCategories(Course $course, array $categories)
+    {
+        $categoriesCollection = Category::whereIn('id', $categories)->get();
+        if ($categoriesCollection) {
+            $courseCategoriesIds = $course->categories()->get()->pluck('id')->toArray();
+            foreach ($categoriesCollection as $category) {
```

Comment:
Consider refactoring this to separate function as this deep nesting is hard to read